### PR TITLE
[stable/janusgraph] Replace deployment with statefulset

### LINKED ISSUE #9663: Cannot scale JanusGraph due to use of PVC within Deployment
BUG REPORT

**Which chart**:
janusgraph

**What happened**:
When attempting to scale the deployment, a second instance fails due to the single PVC.

**What you expected to happen**:
To be able to scale janusgraph. 

**How to reproduce it** (as minimally and precisely as possible):
Increase the `replicaCount` in `values.yaml` and install the chart

**Anything else we need to know**:
The Helm chart should be using a StatefulSet instead of a Deployment to allow scaling with PVCs. See: https://github.com/helm/charts/issues/1863

### PR DESCRIPTION
**Pull request by idcrosby**
#### What this PR does / why we need it:

As per https://github.com/helm/charts/issues/1863. Replaces the Deployment with a StatefulSet. This chart uses PersistentVolumeClaims, scaling the number of replicas currently results in an error.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes https://github.com/helm/charts/issues/9663

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/janusgraph/templates/statefulset.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-apiVersion: apps/v1beta2
-kind: Deployment
+apiVersion: apps/v1beta1
```
Comment (unguiculus) [inline stable/janusgraph/templates/statefulset.yaml:1]: You should go for `apps/v1`.

File: stable/janusgraph/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: "1.0"
 description: Open source, scalable graph database.
 name: janusgraph
-version: 0.2.0
+version: 0.2.1
```
Comment (unguiculus) [inline stable/janusgraph/Chart.yaml:5]: You should bump to `1.0.0`.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Can you elaborate why this change makes sense? I agree that scaling up a deployment with a PVC won't work. The question is how/whether Janusgraph itself can be scaled. If you simply switch to a StatefulSet, you'll get independent replicas with their own database. I would expect that you will somehow have to form a cluster which is not handled by your PR.

**Comment by viglesiasce** [comment]
cc @samizuh

**Comment by samizuh** [comment]
The chart was created so that the default experience was written as a standalone getting started JanusGraph experience that uses Lucene for indexing. The PVC is there so that if you want to preserve your db you can, but the default is for it to be ephemeral. For a scalable solution that can span more than one pod, you need change the indexing backend to use Solr or Elasticsearch as documented [here](https://docs.janusgraph.org/0.2.0/index-backends.html).

**Comment by idcrosby** [comment]
Yes, that makes sense. I had misunderstood how Janus should be configured. We did not need the local persistence (using Elasticsearch). 

The comment on this line threw me off initially: https://github.com/helm/charts/blob/8d8f98816290c64ccda3bac64be95d8b48613d11/stable/janusgraph/values.yaml#L71 

I'll close the PR. Thanks for the responses.

**Comment by samizuh** [comment]
Sounds good. And thank you for catching the typo! (will fix)

**Pull request closed without merging**